internal/deviations/deviations.go

```diff
@@ -12,22 +12,25 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package deviations defines the arguments to enable temporary workarounds
-// for the featureprofiles test suite.
+// Package deviations defines the arguments to enable temporary workarounds for the
+// featureprofiles test suite.
 //
-// Deviations may be introduced to temporarily work around non-compliant issues
-// so further sub-tests can be implemented.  Deviations should be
-// small in scope, typically affecting one sub-test, one OpenConfig
-// path or small OpenConfig sub-tree.  Deviations are enabled using
-// command line parameters.
+// Deviations may be introduced to temporarily work around non-compliant issues so further
+// sub-tests can be implemented.  Deviations should be small in scope, typically affecting
+// one sub-test, one OpenConfig path or small OpenConfig sub-tree.  Deviations are enabled
+// using command line parameters.
 //
-// Passing with a deviation enabled is considered non-compliant to the
-// OpenConfig featureprofiles test.
+// If a device only passes a test by setting the deviation to true, that is considered
+// non-compliant behavior.  Conversely, if a device only passes a test by setting the
+// deviation to false, that is also non-compliant.  Compliant devices must pass with the
+// deviation flags set to either true or false; failing in either case is non-compliant.
```

Comment:
I think a device is compliant if it passes without deviation.  Enabling a deviation I think should mean the test results as non-compliant with the featureprofiles test requirements.  In the future, we can introduce properties to the test log output to programmatically indicate that a deviation is enabled.

I can see scenarios where a compliant device cannot pass with a deviation enabled. Any deviation defined which requires a non-compliant behavior would fall into this set of scenarios.  For example, usage of the default network instance name.  A device can only have one default name for an instance and it must be 'DEFAULT'.  If there is a deviation enabled to define the default network instance name to 'default' and the test is run with this deviation enabled, a compliant device would fail.

Reply:
I didn't mention default VRF because it's not a boolean flag, but I can see why someone might confuse the wording to construe not being able to use a different name for default VRF as non-compliant. I have adjusted the wording in the help string to clarify this particular exception, but I'm also open to renaming `--deviation_default_network_instance` because it is impossible to create a compliant workaround with a different value.

The main idea is that workarounds should only use compliant behavior, and I would like to see tests adhere to this principle unless there is a good reason.

Reply:
Upon closer reading of the standard:

https://github.com/openconfig/public/blob/master/release/models/network-instance/openconfig-network-instance.yang#L1178-L1185

It looks like not being able to use an operator assigned value "default" or otherwise is non-compliant. So we don't have to make this an exception after all.